refactored logic to check for 0 to set proper am/pm period

nick-mon1 opened the pull request:
Updated `themes/digital.gov/src/js/common/github.js` logic to check for 0 hours to assign proper AM/PM periods correctly as identified at https://digital.gov/2016/07/21/confirming-the-cancellation-a-vha-ab-testing-quick-study/

Resolves task https://trello.com/c/ZD8dKLeQ

Test code to check for proper date display:

> <!DOCTYPE html>
> <html lang="en" dir="ltr">
>   <head>
>     <meta charset="utf-8">
>   </head>
>   <body>
> 
>     <script>
> 
>       function getFormattedDate(d) {
>     		var date = new Date(d);
>     		date.setUTCHours(date.getUTCHours() - 4);
>     		var monthNames = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
>     	  var year = date.getUTCFullYear();
>     	  var month = (date.getUTCMonth()).toString();
>     	  month = monthNames[month];
>     	  var day = date.getUTCDate().toString();
>     	  day = day.length > 1 ? day : '0' + day;
>     		var globalhours = parseInt(date.getUTCHours().toString());
>     		if (globalhours === 0) {
>     			var hours = 12;
>     		} else if (globalhours > 12 ) {
>     			var hours = globalhours - 12;
>     		} else {
>     			var hours = globalhours;
>     		}
>     		var minutes = date.getUTCMinutes().toString();
>     		minutes = minutes.length > 1 ? minutes : '0' + minutes;
>     		var seconds = date.getUTCSeconds().toString();
>     		if (globalhours > 12 ) {
>     			var ampm = 'pm';
>     		} else {
>     			var ampm = 'am';
>     		}
>     		var date_string = month + ' ' + day + ', ' + year + ' at ' + hours + ':' + minutes + ' ' + ampm + ' ET';
>         console.log(date_string)
>     	  return date_string;
>     	}
>       getFormattedDate('2022-06-14T23:59:00-04:00');
>       getFormattedDate('2022-06-15T00:00:00-04:00');
>       getFormattedDate('2022-06-15T00:01:00-04:00');
>       getFormattedDate('2022-06-15T00:59:00-04:00');
>       getFormattedDate('2022-06-15T01:00:00-04:00');
>       getFormattedDate('2022-06-15T01:01:00-04:00');
>       getFormattedDate('2022-06-15T11:59:00-04:00');
>       getFormattedDate('2022-06-15T12:00:00-04:00');
>       getFormattedDate('2022-06-15T12:01:00-04:00');
>       getFormattedDate('2022-06-15T12:14:00-04:00');
>       getFormattedDate('2022-06-15T23:59:00-04:00');
>       getFormattedDate('2022-06-16T00:14:00-04:00');
>       getFormattedDate('2022-06-16T00:59:00-04:00');
>     </script>
>   </body>
> </html>
>

mejiaj reviewed (commented):
Nice find, and thanks for writing out examples to help test. It's very helpful!

Just a question, with this test case
```
getFormattedDate('2022-06-15T12:00:00-04:00');
```

I get:
```
Jun 15, 2022 at 12:00 am ET
```

Which is the same I was getting when testing

```
getFormattedDate('2022-06-15T00:00:00-04:00');
```


Should this be 12:00 pm? Maybe I'm misunderstanding

mejiaj commented on themes/digital.gov/src/js/common/github.js:
```diff
@@ -113,6 +115,4 @@ jQuery(document).ready(function($) {
 		var date_string = month + ' ' + day + ', ' + year + ' at ' + hours + ':' + minutes + ' ' + ampm + ' ET';
```
```suggestion
		var date_string = `${month} ${day}, ${year} at ${hours}:${minutes} ${ampm} ET`;
```

nick-mon1 commented:
I've updated the `date_string` formatting and fixed the AM/PM period logic to display correctly for the military time conversion.

> <!DOCTYPE html>
> <html lang="en" dir="ltr">
>   <head>
>     <meta charset="utf-8">
>   </head>
>   <body>
> 
>     <script>
> 
>       function getFormattedDate(d) {
>     		var date = new Date(d);
>     		date.setUTCHours(date.getUTCHours() - 4);
>     		var monthNames = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
>     	  var year = date.getUTCFullYear();
>     	  var month = (date.getUTCMonth()).toString();
>     	  month = monthNames[month];
>     	  var day = date.getUTCDate().toString();
>     	  day = day.length > 1 ? day : '0' + day;
>     		var globalhours = parseInt(date.getUTCHours().toString());
>     		if (globalhours === 0) {
>     			var hours = 12;
>     		} else if (globalhours > 12 ) {
>     			var hours = globalhours - 12;
>     		} else {
>     			var hours = globalhours;
>     		}
>     		var minutes = date.getUTCMinutes().toString();
>     		minutes = minutes.length > 1 ? minutes : '0' + minutes;
>     		var seconds = date.getUTCSeconds().toString();
>     		if (globalhours > 11 ) {
>     			var ampm = 'pm';
>     		} else {
>     			var ampm = 'am';
>     		}
> 
>         var date_string = `${month} ${day}, ${year} at ${hours}:${minutes} ${ampm} ET`;
>         console.log(date_string)
>     	  return date_string;
>     	}
>       getFormattedDate('2022-06-14T23:59:00-04:00');
>       getFormattedDate('2022-06-15T00:00:00-04:00');
>       getFormattedDate('2022-06-15T00:01:00-04:00');
>       getFormattedDate('2022-06-15T00:59:00-04:00');
>       getFormattedDate('2022-06-15T01:00:00-04:00');
>       getFormattedDate('2022-06-15T01:01:00-04:00');
>       getFormattedDate('2022-06-15T11:59:00-04:00');
>       getFormattedDate('2022-06-15T12:00:00-04:00');
>       getFormattedDate('2022-06-15T12:01:00-04:00');
>       getFormattedDate('2022-06-15T13:00:00-04:00');
>       getFormattedDate('2022-06-15T23:59:00-04:00');
>       getFormattedDate('2022-06-16T00:00:00-04:00');
>       getFormattedDate('2022-06-16T12:00:00-04:00');
>     </script>
>   </body>
> </html>

mejiaj reviewed (approved):
LGTM, thanks!
